### PR TITLE
fix: ensure legal identifier for require binding

### LINKED ISSUE #3056: [Bug]: Bad require for code splitting on cjs and arbitrary filename
### Reproduction link or steps

Apply the patch, and run `cargo test --package rolldown --test integration_rolldown -- require_shared`

```diff
From d72059da3e2cdd7720e64b4080ef240da6d31b57 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E4=B8=89=E5=92=B2=E6=99=BA=E5=AD=90=20Kevin=20Deng?=
 <sxzz@sxzz.moe>
Date: Fri, 6 Dec 2024 00:48:35 +0800
Subject: [PATCH] test: require_shared

---
 .../code_splitting/require_shared/_config.json   | 16 ++++++++++++++++
 .../code_splitting/require_shared/main1.js       |  3 +++
 .../code_splitting/require_shared/main2.js       |  3 +++
 .../code_splitting/require_shared/shared-file.js |  1 +
 4 files changed, 23 insertions(+)
 create mode 100644 crates/rolldown/tests/rolldown/code_splitting/require_shared/_config.json
 create mode 100644 crates/rolldown/tests/rolldown/code_splitting/require_shared/main1.js
 create mode 100644 crates/rolldown/tests/rolldown/code_splitting/require_shared/main2.js
 create mode 100644 crates/rolldown/tests/rolldown/code_splitting/require_shared/shared-file.js

diff --git a/crates/rolldown/tests/rolldown/code_splitting/require_shared/_config.json b/crates/rolldown/tests/rolldown/code_splitting/require_shared/_config.json
new file mode 100644
index 000000000..e75e7c18e
--- /dev/null
+++ b/crates/rolldown/tests/rolldown/code_splitting/require_shared/_config.json
@@ -0,0 +1,16 @@
+{
+  "snapshotOutputStats": true,
+  "config": {
+    "input": [
+      {
+        "name": "main1",
+        "import": "main1.js"
+      },
+      {
+        "name": "main2",
+        "import": "main2.js"
+      }
+    ],
+    "format": "cjs"
+  }
+}
diff --git a/crates/rolldown/tests/rolldown/code_splitting/require_shared/main1.js b/crates/rolldown/tests/rolldown/code_splitting/require_shared/main1.js
new file mode 100644
index 000000000..43f5a9092
--- /dev/null
+++ b/crates/rolldown/tests/rolldown/code_splitting/require_shared/main1.js
@@ -0,0 +1,3 @@
+import shared from "./shared-file";
+
+console.log(shared);
diff --git a/crates/rolldown/tests/rolldown/code_splitting/require_shared/main2.js b/crates/rolldown/tests/rolldown/code_splitting/require_shared/main2.js
new file mode 100644
index 000000000..43f5a9092
--- /dev/null
+++ b/crates/rolldown/tests/rolldown/code_splitting/require_shared/main2.js
@@ -0,0 +1,3 @@
+import shared from "./shared-file";
+
+console.log(shared);
diff --git a/crates/rolldown/tests/rolldown/code_splitting/require_shared/shared-file.js b/crates/rolldown/tests/rolldown/code_splitting/require_shared/shared-file.js
new file mode 100644
index 000000000..0b25d9eb7
--- /dev/null
+++ b/crates/rolldown/tests/rolldown/code_splitting/require_shared/shared-file.js
@@ -0,0 +1 @@
+export default 'shared';
\ No newline at end of file
-- 
2.47.1

```

### What is expected?

Pass unit test

### What is actually happening?

```
/home/rolldown/crates/rolldown/tests/rolldown/code_splitting/require_shared/dist/main1.js:2
const require_shared-file = require('./shared-file.js');
      ^^^^^^^^^^^^^^

SyntaxError: Missing initializer in const declaration
```

### System Info

```Shell
N/A
```

### Any additional comments?

I'll try to fix it

### PR DESCRIPTION
**Pull request by sxzz**
### Description

closes #3056

### CODE REVIEW

File: crates/rolldown_utils/src/sanitize_identifier.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+use std::borrow::Cow;
+
+use oxc::syntax::keyword::RESERVED_KEYWORDS;
+
+// Follow from https://github.com/rollup/rollup/blob/master/src/utils/identifierHelpers.ts#L17-L25
+#[allow(clippy::needless_pass_by_value)]
+pub fn sanitize_identifier(str: Cow<str>) -> String {
```
Comment (IWANABETHATGUY) [inline crates/rolldown_utils/src/sanitize_identifier.rs]: There already exists a util function here https://github.com/rolldown/rolldown/blob/2ec340edda3da25c3be15562335b62d1a8a787f8/crates/rolldown_utils/src/ecmascript.rs?plain=1#L14-L49

Comment (sxzz) [reply crates/rolldown_utils/src/sanitize_identifier.rs]: Thanks! I'm looking for it

**Comment by hyf0** [comment]
Run `just test-update` to update all snapshots.

**Review by hyf0 — APPROVED** [review]
Thanks

**Pull request merged**